### ISSUE TITLE
Consider wrapping group dropdown and sidebar toolbar in small resolution

### ISSUE THREAD

**Issue by acelaya** [hypothesis/client#7315]
As part of https://github.com/hypothesis/product-backlog/issues/1660, we have applied some layout improvements that address the lack of space in the sidebar's top bar.

However, when the sidebar is resized to be very small, all the components in the top bar have to share very little space, which causes the group name to be truncated in a way that very little is seen.

We want to explore the option of making the group dropdown and the sidebar toolbar wrap into two different rows for small resolutions, but this requires experimenting a bit and seeing how it affects the rest of the sidebar layout.
